packages/entity/src/EntityQueryContext.ts

```diff
@@ -46,10 +50,20 @@ export class EntityTransactionalQueryContext extends EntityQueryContext {
   private readonly postCommitInvalidationCallbacks: PostCommitCallback[] = [];
   private readonly postCommitCallbacks: PostCommitCallback[] = [];
 
+  private readonly preCommitCallbacks: { callback: PreCommitCallback; order: number }[] = [];
+
+  /**
+   * Schedule a pre-commit callback. These will be run within the transaction.
+   * @param callback - callback to schedule
+   * @param order - order in which this should be run, with higher numbers running later than lower numbers
```

Comment:
How do people choose what number to specify for the order? Some guidance would be helpful if the intent is, "I don't care as long as the callback runs before PG's COMMIT" -- should order be `0`? `MIN_SAFE_INTEGER`? `MAX_SAFE_INTEGER`?